projects/pt1/e2e_testing/main.py

```diff
@@ -152,21 +162,24 @@ def main():
         if re.match(args.filter, test.unique_name)
     ]
     if len(tests) == 0:
-        print(
+        logger.error(
             f"ERROR: the provided filter {args.filter!r} does not match any tests"
         )
-        print("The available tests are:")
-        for test in available_tests:
-            print(test.unique_name)
+        available_tests = [test for test in available_tests]
+
+        logger.error("The available tests are:" + "\n".join(available_tests))
         sys.exit(1)
 
     # Run the tests.
-    results = run_tests(tests, config, args.sequential, args.verbose)
+    results = run_tests(tests, config, args.sequential,
+                            verbose=logger.level >= logging.INFO)
 
     # Report the test results.
-    failed = report_results(results, xfail_set, args.verbose, args.config)
+    failed = report_results(results, xfail_set,
+                            verbose=logger.level >= logging.INFO,
+                            config=args.config)
     if args.config == "torchdynamo":
-        print("\033[91mWarning: the TorchScript based dynamo support is deprecated. "
+        logger.warning("\033[91mWarning: the TorchScript based dynamo support is deprecated. "
               "The config for torchdynamo is planned to be removed in the future.\033[0m")
```

Comment:
Similar comment here - the escape code + "Warning:" text may be redundant with the logging level already set to 'warning'.

Reply:
fixd